File: web/src/app/core/common/paged-result.ts
/** Mirrors ErpApp.Application.Common.Pagination.PagedResult<T> -- the one wire shape every
 * retrofitted list/report endpoint returns (roadmap Phase 16c). TotalPages/hasNext are derived
 * client-side, matching the server's own "don't model what the client can compute" choice. */
export interface PagedResult<T> {
  items: T[];
  page: number;
  pageSize: number;
  totalCount: number;
}

export const DEFAULT_PAGE_SIZE = 50;

/** Matches ErpApp.Application.Common.Pagination.PagingDefaults.MaxPageSize -- used by dropdown/
 * picker call sites (e.g. populating a Customer/Product/Account <select>) that need "give me
 * everything" rather than a paginated slice. Not a full fix for a tenant with more rows than this
 * in a single picker (that needs a real searchable async select, a separate future concern) but
 * avoids silently truncating today's realistic org sizes to the 50-row list-page default. */
export const MAX_PAGE_SIZE = 200;

File: src/Application/Common/Pagination/PagedResult.cs
namespace ErpApp.Application.Common.Pagination;

/// <summary>
/// Uniform envelope for every paginated list/report query (roadmap Phase 16c, NFR-5.1) -- one
/// shared type used across all 30 retrofitted queries rather than a per-module duplicate, so the
/// Angular side has exactly one shape to unwrap. TotalPages/HasNextPage are deliberately not
/// modeled server-side -- they're pure derivations of Page/PageSize/TotalCount the client can
/// compute itself.
/// </summary>
public sealed record PagedResult<T>(IReadOnlyList<T> Items, int Page, int PageSize, int TotalCount);

/// <summary>
/// Shared paging bounds enforced by every retrofitted query's FluentValidation validator (see
/// PagingValidation.ValidatePaging) -- Page &lt; 1 or PageSize outside [1, MaxPageSize] is
/// rejected with a 400, never silently clamped (phase-16c-status.md's boundary-correctness exit
/// criterion). DefaultPageSize matches the pre-existing unpaginated screens' typical row count
/// closely enough that a caller which omits both params sees a page size that still feels
/// "give me everything" for a small tenant.
/// </summary>
public static class PagingDefaults
{
    public const int DefaultPageSize = 50;
    public const int MaxPageSize = 200;
}
